core/node/libp2p/rcmgr_defaults.go

```diff
@@ -186,5 +186,23 @@ Run 'ipfs swarm limit all' to see the resulting limits.
 
 	defaultLimitConfig := scalingLimitConfig.Scale(int64(maxMemory), int(numFD))
 
+	// Simple checks to overide autoscaling ensuring limits make sense versus the connmgr values.
+	// There are ways to break this, but this should catch most problems already.
+	// We might improve this in the future.
+	// See: https://github.com/ipfs/kubo/issues/9545
+	if cfg.ConnMgr.Type == nil || cfg.ConnMgr.Type.String() != "none" {
+		maxInboundConns := int64(defaultLimitConfig.System.ConnsInbound)
+		if connmgrHighWaterTimesTwo := cfg.ConnMgr.HighWater.WithDefault(config.DefaultConnMgrHighWater) * 2; maxInboundConns < connmgrHighWaterTimesTwo {
+			maxInboundConns = connmgrHighWaterTimesTwo
+		}
+		// MAGIC: 800 is probably a good enough number to be a good network citizen.
+		if maxInboundConns < 800 {
+			maxInboundConns = 800
+		}
```

Comment:
Move magic number  `800` to be a `const DefaultResourceMgrMinInboundConns` in `init.go`  (we already have `DefaultConnMgrLowWater` there)

Reply:
Done